x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -411,4 +411,9 @@ public void onFailure(Exception e) {
             }
         });
     }
+
+    @FunctionalInterface
+    private interface MatchFieldMapping {
+        XContentBuilder build(XContentBuilder builder) throws IOException;
```

Comment:
Maybe just use `CheckedFunction<XContentBuilder, XContentBuilder, IOException >` instead?
Adding a functional interface just for resolving mapping in a private method seems a bit too much?

Reply:
oh woah, I had no idea CheckedFunction existed! yes, I will use that.